annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,223 @@
+
+            val annotationSheet: MutableList<Annotation> = arrayListOf()
+
+            // Go through each file in each user's folder and collect the stats
+            var user: String? = null
+            var subtype: String? = null
+            File(exportAnnotationRoot.toString()).walk().forEach {
+                // Skip irrelevant dirs
+                if (it == File(exportAnnotationRoot.toString()) ||
+                        it.name.contains("""sandbox""") ||
+                        it.name.contains("""gigaword""")) {
+                }
+                else if (it.isDirectory) {
+                    val folder = it.name
+                    // Collect username, subtype, and subSubtype
+                    user = Regex(pattern = """\.\w*-(\w*)""")
+                            .find(input = folder.toString())!!.groupValues[1]
+                    subtype = Regex(pattern = """(\w*\.\w*)-""")
+                            .find(input = folder.toString())!!.groupValues[1]
+                }
+                else if (it.isFile) {
+                    if (user.isNullOrBlank() || subtype.isNullOrBlank()) {
+                        throw RuntimeException("Missing annotation information (user/subtype)")
+                    }
+                    val documentName: String = it.toString()
+                    // TODO: count events rather than annotated documents
+                    // This is sort of a placeholder for a better method
+                    val jsonString: String = File(it.toString()).readText(Charsets.UTF_8)
+                    if (jsonString.contains(Regex(""""CTEventSpan" : \["""))) {
+                        annotationSheet.add(Annotation(documentName, user!!, subtype!!))
+                    }
+                }
+            }
+            // From annotationSheet, get TOTAL, TOTAL PER USER/SUBTYPE/SUBSUBTYPE
+            val totalAnnotations = annotationSheet.size
+            val annotationsPerUser = annotationSheet.countBy { it.user }
+            val annotationsPerSubtype = annotationSheet.countBy { it.subtype }
+            // For better organization
+            val sortedUsers = annotationsPerUser.toSortedMap()
+            val sortedSubtypes = annotationsPerSubtype.toSortedMap()
+
+            val lastFilename = locatePreviousStats(currentDate, statisticsDirectory)
+            val previousAnnotationsPerUser: MutableMap<String, Int> = mutableMapOf()
+            val previousAnnotationsPerSubtype: MutableMap<String, Int> = mutableMapOf()
+            var previousAnnotations = 0
+            if (lastFilename.isEmpty()) {
+                logger.info {"No previous stats report in over a week"}
+            }
+            else {
+                previousAnnotations = getPreviousAnnotations(File(lastFilename), previousAnnotationsPerUser, previousAnnotationsPerSubtype)
+            }
+
+            // Convert values to html
+            File("$statisticsDirectory/StatsReport$thisDate.html").printWriter().use{
```

Comment:
prefer
```
File(statisticsDirectory, "StatsReport$thisDate.html")
```

(again, avoid working with strings instead of using the provided logic)